src/org/zaproxy/zap/extension/websocket/manualsend/ManualWebSocketSendEditorDialog.java

```diff
@@ -261,5 +273,72 @@ public void addSeparator() {
 		public void addEndButton(JButton button) {
 			messagePanel.addOptions(button, HttpPanel.OptionsLocation.END);
 		}
+
+		private JButton getBtnReopen(WebSocketPanelSender sender) {
+			if (btnReopen == null) {
+				btnReopen = new JButton();
+				btnReopen.setIcon(reopenIcon);
+				btnReopen.setEnabled(true);
+				btnReopen.setToolTipText(Constant.messages.getString("websocket.manual_send.btn_reopen_hint"));
+				btnReopen.addActionListener(e -> {
+					if(wsUiHelper.isAlwaysGenerate()){
+						newSecWebSocketKey = WebSocketUtils.generateSecWebSocketKey();
+					}else {
+						newSecWebSocketKey = ((wsUiHelper.isGenerateKeyPressed()) ? wsUiHelper.getWebSocketKey() : null);
```

Comment:
I think there are some extra brackets here. Likely syntactically correct but probably not necessary. 

`int someNum = ((str.isEmpty()) ? 0 : 10);`
should be the same as
`int someNum = str.isEmpty() ? 0 : 10;`